spec/models/miq_queue_spec.rb

```diff
@@ -986,4 +986,60 @@ def queue_items
       )
     end
   end
+
+  describe ".messaging_client_options" do
+    before do
+      config_file = Rails.root.join("config", "messaging.yml")
+      FileUtils.cp(Rails.root.join("config", "messaging.kafka.yml"), config_file) unless config_file.file?
+    end
+
+    context "with ENV" do
+      around do |example|
+        old_env = ENV
+        ENV["MESSAGING_HOSTNAME"] = "server.example.com"
+        ENV["MESSAGING_PASSWORD"] = "password"
+        ENV["MESSAGING_PORT"]     = "9092"
+        ENV["MESSAGING_USERNAME"] = "admin"
```

Comment:
Can we use stub_const for this instead?  If so, then this becomes

```ruby
before do
  stub_const("ENV", ENV.to_h.merge(
    "MESSAGING_HOSTNAME" => "server.example.com"
    "MESSAGING_PASSWORD" => "password"
    "MESSAGING_PORT"     => "9092"
    "MESSAGING_USERNAME" => "admin"
  ))
end

Reply:
Done